### ISSUE TITLE
node.js minor version not proposed

### ISSUE THREAD

**Issue by bodinsamuel** [renovatebot/config-help#813]
**What Renovate type are you using?**

Hosted app for github

**Describe the bug**

- No new version for node.js is proposed 
The .nvm is at 12.16.2 I would expect to have update proposed for [12.18.3](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md#12.18.3)
Instead I see that renovate is trying to upgrade to 14.8.0

Can you provide some guidance?
Thanks

**Relevant debug logs**

```js
{
            "major": {
              "enabled": false
            },
            "depName": "node",
            "depType": "engines",
            "updates": [
              {
                "fromVersion": "12.16.2",
                "toVersion": "v14.8.0",
                "newValue": "14.8.0",
                "newMajor": 14,
                "newMinor": 8,
                "updateType": "major",
                "isSingleVersion": true
              }
            ],
            "depIndex": 37,
            "warnings": [],
            "sourceUrl": "https://github.com/nodejs/node",
            "datasource": "github-tags",
            "lookupName": "nodejs/node",
            "versioning": "node",
            "currentValue": "12.16.2",
            "prettyDepType": "engine",
            "commitMessageTopic": "Node.js"
          },
```

**To Reproduce**
```json
{
  "extends": [
    ":docker",
    ":separateMajorReleases",
    ":combinePatchMinorReleases",
    ":ignoreUnstable",
    ":semanticPrefixFixDepsChoreOthers",
    ":noUnscheduledUpdates",
    ":ignoreModulesAndTests",
    ":autodetectPinVersions",
    "group:monorepos",
    "group:recommended"
  ],
  "enabledManagers": [
    "circleci",
    "docker-compose",
    "dockerfile",
    "kubernetes",
    "npm",
    "nvm"
  ],
  "automerge": true,
  "separateMultipleMajor": true,
  "schedule": "every weekend",
  "reviewers": [],
  "semanticCommitType": "chore",
  "baseBranches": [
    "chore/renovateBaseBranch"
  ],
  "docker-compose": {
    "ignoreDeps": [
      "kibana-oss",
      "elasticsearch-oss"
    ]
  },
  "node": {
    "supportPolicy": [
      "lts_active"
    ]
  },
  "kubernetes": {
    "enabled": true,
    "fileMatch": [
      "kubernetes/templates/.+\\.yml$"
    ]
  },
  "lockFileMaintenance": {
    "enabled": false
  },
  "masterIssue": true,
  "stabilityDays": 7,
  "prCreation": "not-pending",
  "timezone": "Europe/Paris",
  "prHourlyLimit": 5,
  "prConcurrentLimit": 10
}

```

**Additional context**

**Comment by bodinsamuel** [comment]
oops yes it was the log for the engine field in the package.json, but I guess the resolver is the same

**Comment by rarkins** [comment]
Can you recreate the problem in a public repo?

**Comment by bodinsamuel** [comment]
I removed most of the config, still no update.
latest job 220843958

**Comment by rarkins** [comment]
I think I see is not. The problem is this config: `"automerge": true,`

Setting it in the root means "automerge for all releases, even major".

It also means that internally, we don't propose multiple updates for minor and major anymore, because if you're going to automerge the major update anyway then there's no point in having a race condition minor update at the same time.

This then kind of conflicts with the fact that we don't want to propose node v14, the end result being no updates.

In short:
- automerge=true means that if a major update is present then no minor update will be proposed
- node v14 is then skipped
- therefore, no node v12

I suggest adding [:automergeMinor](https://docs.renovatebot.com/presets-default/#automergeminor) to your extends and removing `automerge=true` from your config.

**Comment by bodinsamuel** [comment]
Fair enough, but that would be cool to report it either on the PR or on the master issue, because the log are fairly unreadable.

Now that you mention it I have another feedback, since last week `automerge` seems to not work or at least something changed. On our repo all PRs now requires to be approved.
The config is the same as the initial issue.

```
[...]
Branch protection: PR Reviews are required before merging(branch="renovate/docker-datadog-agent-7.x")
```

**Comment by rarkins** [comment]
> On our repo all PRs now requires to be approved.

That's a GitHub setting, not Renovate. Most likely someone in your org enabled that in Branch protection. You can install https://github.com/apps/renovate-approve as a second bot that approves all of Renovate's automerging PRs to workaround this protection, if desired.

**Comment by bodinsamuel** [comment]
> That's a GitHub setting, not Renovate. Most likely someone in your org enabled that in Branch protection. You can install https://github.com/apps/renovate-approve as a second bot that approves all of Renovate's automerging PRs to workaround this protection, if desired.

Our branch protection is only on `pr > master`, renovate will create against baseBranch which is `chore/renovateBaseBranch` and this one is not protected.

**Comment by rarkins** [comment]
I created https://github.com/renovatebot/renovate/issues/7077 to address that, so let's close this one
